gnomad/utils/constraint.py

```diff
@@ -226,6 +226,31 @@ def count_variants_by_group(
         )
 
 
+def get_downsampling_freq_indices(
+    freq_meta_expr: hl.expr.ArrayExpression,
+    pop: str = "global",
+    variant_quality: str = "adj",
+) -> hl.expr.ArrayExpression:
+    """
+    Get indices of dictionaries in meta dictionaries that only have the "downsampling" key with specified "pop" and "variant_quality" values.
+
+    :param freq_meta_expr: ArrayExpression containing the set of groupings for each
+        element of the `freq_expr` array (e.g., [{'group': 'adj'}, {'group': 'adj',
+        'pop': 'nfe'}, {'downsampling': '5000', 'group': 'adj', 'pop': 'global'}]).
+    :param pop: Population to use for filtering by the 'pop' key in `freq_meta_expr`.
+        Default is 'global'.
+    :param variant_quality: Variant quality to use for filtering by the 'group' key in
+        `freq_meta_expr`. Default is 'adj'.
+    """
+    indices = hl.enumerate(freq_meta_expr).filter(
+        lambda f: (f[1].get("group") == variant_quality)
```

Comment:
Just confirming `lambda f: (f[1].size() == 3)` is not needed. There is no case where there will be something like: `{'downsampling': '5000', 'group': 'adj', 'pop': 'global'}` and `{'downsampling': '5000', 'group': 'adj', 'pop': 'global', 'other_strata':'some_val}` and we only want `{'downsampling': '5000', 'group': 'adj', 'pop': 'global'}`?

Reply:
at least not in the existing datasets, but I could leave it in to be on the safe size

Reply:
I think it's fine for now and we can update if our downsampling frequencies get more complex